### PR TITLE
Print requests latency

### PR DESCRIPTION
**Pull request by kirpichenko**

### CODE REVIEW

File: packages/grpcts/src/LatencyTimer.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  public start(): Latency {
+    const now = new Date();
+    return new Latency(now);
+  }
+}
+
+class Latency {
+  constructor(readonly startDate: Date) {}
+
+  public getValue(): number {
```
Comment (minaeiehsan) [inline packages/grpcts/src/LatencyTimer.ts]: don't like a getter method?
`get value() {} ...`

File: packages/grpcts/src/Client.ts [hunk 2]
```diff
@@ -9,14 +10,20 @@ export interface ClientTrace {
...
   public client: GrpcClient;
   constructor(
-    definition: grpc.ServiceDefinition<any>,
+    readonly definition: grpc.ServiceDefinition<any>,
     address: string,
     credentials: grpc.ChannelCredentials = grpc.credentials.createInsecure(),
     public readonly trace?: ClientTrace,
-    options?: object
+    options?: object,
+    private readonly logger?: Logger,
```
Comment (tcichowicz) [inline packages/grpcts/src/Client.ts]: The constructor function signature grew a lot - should we replace it with a single object?

File: packages/grpcts/src/Client.ts [hunk 3]
```diff
@@ -32,18 +39,38 @@ export class Client {
...
       call = this.client[methodName](
         req,
         this.metadata(metadata),
         (err: grpc.ServiceError, res: ResponseType) => {
-          err ? reject(this.convertError(err)) : resolve(res);
+          if (err) {
+            return reject(this.convertError(err));
+          }
+
+          this.log(methodName, req, latency);
```
Comment (tcichowicz) [inline packages/grpcts/src/Client.ts]: We log only latency of successful responses?

File: packages/grpcts/src/Service.ts [hunk 4]
```diff
@@ -56,23 +57,23 @@ const logging = (
...
-  } = {
-    path: definition.path
+  const request = !definition.requestStream ? call.request : 'STREAM';
+  const response = !definition.responseStream ? result : 'STREAM';
+  const isError = result instanceof Error;
+  const logData = {
+    request,
+    [isError ? 'error' : 'response']: response,
+    path: definition.path,
+    emitter: 'service'
```
Comment (tcichowicz) [inline packages/grpcts/src/Service.ts]:
```suggestion
    emitter: 'service',
    latency: latency?.getValue()
```
Would this work?

File: packages/grpcts/src/Client.ts (hunk 2 above)
Comment (kirpichenko) [reply packages/grpcts/src/Client.ts]: i don't really want to deep and refactor it right now. want to check it this actually works or not.

File: packages/grpcts/src/Client.ts (hunk 3 above)
Comment (kirpichenko) [reply packages/grpcts/src/Client.ts]: yes. will adjust for all requests

File: packages/grpcts/src/Service.ts (hunk 4 above)
Comment (kirpichenko) [reply packages/grpcts/src/Service.ts]: true. it was optional from the beginning,  but i added default timer

File: packages/grpcts/src/LatencyTimer.ts (hunk 1 above)
Comment (kirpichenko) [reply packages/grpcts/src/LatencyTimer.ts]: want to leave it as a method, so it's clear form outside that the value is actually calculated when you call it

**Pull request merged**